db/upgrade.php

```diff
@@ -269,5 +225,54 @@ function xmldb_report_customsql_upgrade($oldversion) {
         upgrade_plugin_savepoint(true, 2021111600, 'report', 'customsql');
     }
 
-    return true;
+    if ($oldversion < 2022031801) {
+
+        // Define table report_customsql_categories to be created.
+        $table = new xmldb_table('report_customsql_categories');
+
+        // Adding fields to table report_customsql_categories.
+        $table->add_field('id', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, XMLDB_SEQUENCE, null, null);
+        $table->add_field('name', XMLDB_TYPE_CHAR, '255', null, XMLDB_NOTNULL, null, null, null);
+
+        // If the 'name' field already exists, ensure its constraint is set to NOTNULL
+        if ($dbman->field_exists($table, $table->getField('name'))) {
```

Comment:
Wasn't sure where to put this, this just makes sure that if the column exists already make sure it has the correct constraint defined on [#L235](https://github.com/catalyst/moodle-report_customsql/blob/issue123-categories-schema/db/upgrade.php#L235).

Reply:
Ah, yes the field_exists check is good to have in the new upgrade block.